homeassistant/components/watergate/entity.py

```diff
@@ -0,0 +1,50 @@
+"""Watergate Base Entity Definition."""
+
+from dataclasses import dataclass
+
+from watergate_local_api import WatergateLocalApiClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import WatergateDataCoordinator
+
+type WatergateConfigEntry = ConfigEntry[WatergateData]
+
+
+class WatergateEntity(CoordinatorEntity[WatergateDataCoordinator]):
+    """Define a base Watergate entity."""
+
+    def __init__(
+        self,
+        coordinator: WatergateDataCoordinator,
+        entry: WatergateConfigEntry,
```

Comment:
I always prefer not to have the entry in an entity as it feels like a big sluggish object

Reply:
Again since I am going to use the serial number which will be in the coordinator data, then I will just remove it.